overlord/snapstate/snapmgr.go

```diff
@@ -100,117 +107,106 @@ func Manager(s *state.State) (*SnapManager, error) {
 }
 
 func (m *SnapManager) doDownloadSnap(t *state.Task, _ *tomb.Tomb) error {
-	var inst installState
-	var dl downloadState
+	var ss snapSetup
 
 	t.State().Lock()
-	if err := t.Get("install-state", &inst); err != nil {
+	err := t.Get("snap-setup", &ss)
+	t.State().Unlock()
+	if err != nil {
 		return err
 	}
-	t.State().Unlock()
 
 	pb := &TaskProgressAdapter{task: t}
-	downloadedSnapFile, developer, err := m.backend.Download(inst.Name, inst.Channel, pb)
+	downloadedSnapFile, version, err := m.backend.Download(ss.Name, ss.Channel, pb)
 	if err != nil {
 		return err
 	}
-	dl.SnapPath = downloadedSnapFile
-	dl.Developer = developer
+	ss.SnapPath = downloadedSnapFile
+	ss.Version = version
 
-	// update instState for the next task
+	oldInstPath, _ := filepath.EvalSymlinks(filepath.Join(ss.BaseDir(), "..", "current", "meta", "snap.yaml"))
+	if sn, err := snappy.NewInstalledSnap(oldInstPath); err == nil {
```

Comment:
mostly a nitpick: this bits feels a bit out of place in terms of level, like it should be backend operation, this is the only place we use snappy.* to do something serious in this file